vslib/VirtualSwitchSaiInterface.cpp

```diff
@@ -847,6 +847,19 @@ sai_status_t VirtualSwitchSaiInterface::objectTypeGetAvailability(
         *count = 512;
         return SAI_STATUS_SUCCESS;
     }
+    else if ((objectType == (sai_object_type_t)SAI_OBJECT_TYPE_VNET) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_ENI) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_ENI_ETHER_ADDRESS_MAP_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_INBOUND_ROUTING_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_OUTBOUND_ROUTING_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_PA_VALIDATION_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_OUTBOUND_CA_TO_PA_ENTRY) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_DASH_ACL_GROUP) ||
+             (objectType == (sai_object_type_t)SAI_OBJECT_TYPE_DASH_ACL_RULE))
```

Comment:
why those specific types ? do they need to be listed explicitly, seems like all of t hem are experimental objects, maybe you could use compare ?
you could get object_info from sai_metadata_get_object_type_info(ot) (check for null) on top of the function and then use info->isexperimental flag,
or if objectType = SAI_OBJECT_TYPE_MAX

Reply:
those are the DASH objects that we use and want to test in CRM DASH VS tests. Not all experimental objects are resources, so info->isexperimental flag and objectType >= SAI_OBJECT_TYPE_MAX are redundant.

Reply:
then use flag md->isresourcetype ?